core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -146,6 +148,14 @@ final class ShuffleBlockFetcherIterator(
   /** Current number of blocks in flight per host:port */
   private[this] val numBlocksInFlightPerAddress = new HashMap[BlockManagerId, Int]()
 
+  /**
+   * The average size of the remote blocks. We'll only unset `isNettyOOMOnShuffle` when
+   * the free Netty memory is larger than this.
+   *
+   * Visible to [[BufferReleasingInputStream]]
+   */
+  var averageRemoteBlockSize = 0L
```

Comment:
does it need to be thread-safe?

Reply:
it doesn't. It's a filed to the shuffle block fetcher iterator instance.